### ISSUE TITLE
Build training dataset (recipe v20) from per-bin segmentation predictions

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#133]
🤖 Sub-issue of #96. Closes the loop between #118 / PR #126 (per-bin segmentation logits across 20 mammals) and an actual gLM training dataset that consumes them.

## Context

PR #126 produces per-bin enhancer logits at 128 bp resolution for 20 diverse mammals, written to
`s3://oa-bolinas/snakemake/training_dataset/dataset_creation/results/enhancer_predictions_segmentation/{g}.parquet`
with schema `(chrom, bin_start, bin_end, logit)`.

Today this output is unused — there is no recipe that turns it into a training dataset. The classifier-based recipe `v19` (in `enhancer_prediction.smk`) only consumes the older `predict_enhancers` per-window logits, and even those aren't wired into `intervals.training` in `config.yaml`.

This issue adds **recipe v20**: a segmentation-prediction-driven enhancer recipe, integrated into the existing dataset-assembly path so it produces an HF training dataset for the same 20 mammals already covered by PR #126. It is the segmentation analogue of `v19`.

## Design decisions

1. **Threshold:** per-genome quantile of logits (label-free). Default `top_quantile = 0.01` (top 1% of bins per genome). No held-out PR-curve calibration in this iteration — that's a follow-up under #96.
2. **Interval form:** each above-threshold 128 bp bin is resized to a 255 bp window centered on its midpoint (mirrors `v17`/`v18`).
3. **Exon subtraction:** subtract **all functional exons** (CDS + UTR + ncRNA — i.e. the union of all `feature == \"exon\"` rows from the GTF, with low-quality biotypes such as NMD / retained intron / pseudogenes already excluded). Reuse the existing `extract_exons` rule output, produced by `get_exons_for_masking` in `src/bolinas/data/utils.py`.
4. **Scope:** all 20 mammals from `enhancer_prediction_segmentation.genomes` in `config.yaml`. New `genome_sets` entry `enhancer_seg_mammals_v1` with explicit accession list, and recipe v20 added to `intervals.training` paired with that set.

## Implementation outline

- New helper `top_quantile_bins_to_windows(bin_logits, top_quantile, target_size) -> GenomicSet` in `src/bolinas/data/bin_predictions.py`, with unit tests.
- New `intervals_recipe_v20` rule in `snakemake/training_dataset/dataset_creation/workflow/rules/enhancer_prediction.smk`:
  ```
  windows = top_quantile_bins_to_windows(predictions, top_quantile, target_size)
  windows = (windows - exons) & defined
  windows.write_bed(...)
  ```
- Extend `load_genome_sets` in `common.smk` to support an explicit `accessions:` list (in addition to the existing `rank_key`/`rank_value`).
- Config additions in `config.yaml`: `recipe_top_quantile`, `recipe_target_size` inside `enhancer_prediction_segmentation`; new `enhancer_seg_mammals_v1` genome set; `v20/255/128` in `intervals.training`.
- README section documenting v20.

## Verification

- `uv run pytest tests/data/test_bin_predictions.py` (local).
- Snakemake `--dry-run` for human and the full v20 target.
- Single-genome smoke test on human (SkyPilot CPU instance): row count, length distribution, exon-overlap = 0, coordinate sanity.
- Full 20-genome run + HF upload: `bolinas-dna/genomes-v5-genome_set-enhancer_seg_mammals_v1-intervals-v20_255_128`.

## Out of scope

- Threshold calibration on chr7 with held-out PR curves (deferred — would replace the quantile threshold with a precision-targeted one).
- Validation split for v20.
- gLM training comparison v18/v19/v20 (tracked under #96).
- Expanding to >20 mammals (blocked on #119).
